ocaml/parsing/pprintast.ml

```diff
@@ -1826,6 +1826,19 @@ and label_x_expression_param ctxt f (l,e) =
       else
         pp f "~%s:%a" lbl (simple_expr ctxt) e
 
+and tuple_component ctxt f (l,e) =
+  let simple_name = match e with
+    | {pexp_desc=Pexp_ident {txt=Lident l;_};
+       pexp_attributes=[]} -> Some l
+    | _ -> None
+  in match l with
+  | None  -> expression2 ctxt f e (* level 2*)
+  | Some lbl ->
+      if Some lbl = simple_name then
+        pp f "~%s" lbl
+      else
+        pp f "~%s:%a" lbl (simple_expr ctxt) e
```

Comment:
Thanks, updated to match on `(simple_name, l)` and use `String.equal`.

I kept the `let simple_name =` line above because I liked separating the logic of reaching into the `expression` and because it matches the pattern used in `label_x_expression_param`, but it wouldn't be a problem to use a nested match if you prefer!